#### Fluoxetine VS Saline for stressed in Dentate Gyrus {#resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrus}

##### Default settings {#resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrusDefaultSettings}

```{r resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrusDefaultSettings}
gset <- readRDS(
  file.path(
    mw_from_rmd,
    "out/r/preprocess_Surget_2009.rds"
  )
)
fData(gset)$ID <- rownames(gset)
fData(gset)$mouse_symbol <- mapIds(
  x = mouse4302.db,
  keys = rownames(gset),
  column = "SYMBOL",
  keytype = "PROBEID"
)

fData(gset) <- cbind(
  fData(gset),
  annotations$AFFY_Mouse430_2[
    match(
      fData(gset)$ID,
      annotations$AFFY_Mouse430_2$probe
    )
  ]
)

gset <- gset[
  ,
  pData(gset)$stress %in% "S" &
  pData(gset)$tissue %in% "DG" &
  pData(gset)$treatment %in% c("FLX","Sal")
]
treatment <- factor(gset[["treatment"]])

design <- model.matrix(~ 0 + treatment)

fit <- lmFit(
  gset,
  design
)

contrast_matrix <- makeContrasts(
  contrasts = "treatmentFLX-treatmentSal",
  levels = design
)
fit2 <- contrasts.fit(
  fit,
  contrast_matrix
)
# compute statistics and table of top significant genes
fit2 <- eBayes(
  fit2,
  0.01
)
tT <- topTable(
  fit2,
  adjust = "fdr",
  sort.by = "B",
  number = 250
)

tT <- subset(
  tT,
  select = c(
    "adj.P.Val",
    "P.Value",
    "t",
    "B",
    "logFC",
    "ID",
    "Gene.stable.ID",
    "Gene.name",
    "mouse_symbol"
  )
)
produceDataTableWithButtons(tT)
# write.table(tT, file = stdout(), row.names = F, sep = "\t")

# Visualize and quality control test results.
# Build histogram of P-values for all genes. Normal test
# assumption is that most genes are not differentially expressed.
tT2 <- topTable(
  fit2,
  adjust = "fdr",
  sort.by = "B",
  number = Inf
)
hist(
  tT2$adj.P.Val,
  col = "grey",
  border = "white",
  xlab = "P-adj",
  ylab = "Number of genes",
  main = "P-adj value distribution"
)

# summarize test results as "up", "down" or "not expressed"
dT <- decideTests(
  fit2,
  adjust.method = "fdr",
  p.value = 0.05
)

# Venn diagram of results
vennDiagram(
  dT,
  circle.col = palette()
)

# create Q-Q plot for t-statistic
t.good <- which(!is.na(fit2$F)) # filter out bad probes
qqt(
  fit2$t[t.good],
  fit2$df.total[t.good],
  main = "Moderated t statistic"
)

# volcano plot (log P-value vs log fold change)
colnames(fit2) # list contrast names
ct <- 1 # choose contrast of interest
volcanoplot(
  fit2,
  coef = ct,
  main = colnames(fit2)[ct],
  pch = 20,
  highlight = length(which(dT[, ct] != 0)),
  names = rep("+", nrow(fit2))
)

# MD plot (log fold change vs mean log expression)
# highlight statistically significant (p-adj < 0.05) probes
plotMD(
  fit2,
  column = ct,
  status = dT[, ct],
  legend = FALSE,
  pch = 20,
  cex = 1
)
abline(h = 0)
################################################################
# General expression data analysis
ex <- exprs(gset)
# box-and-whisker plot
# ord <- order(gs) # order samples by group
# palette(c(
#   "#1B9E77", "#7570B3", "#E7298A", "#E6AB02", "#D95F02",
#   "#66A61E", "#A6761D", "#B32424", "#B324B3", "#666666"
# ))
# par(mar = c(7, 4, 2, 1))
# title <- paste(
#   "Surget 2009",
#   "/",
#   annotation(gset),
#   sep = ""
# )
# boxplot(
#   ex[, ord],
#   boxwex = 0.6,
#   notch = TRUE,
#   main = title,
#   outline = FALSE,
#   las = 2,
#   col = gs[ord]
# )
# legend(
#   "topleft",
#   groups,
#   fill = palette(),
#   bty = "n"
# )
# # expression value distribution
# par(mar = c(4, 4, 2, 1))
# title <- paste(
#   "Surget 2009",
#   "/",
#   annotation(gset),
#   " value distribution",
#   sep = ""
# )
# plotDensities(
#   ex,
#   group = gs,
#   main = title,
#   legend = "topright"
# )
# # UMAP plot (dimensionality reduction)
# ex <- na.omit(ex) # eliminate rows with NAs
# ex <- ex[!duplicated(ex), ] # remove duplicates
# ump <- umap(t(ex), n_neighbors = 4, random_state = 123)
# par(mar = c(3, 3, 2, 6), xpd = TRUE)
# plot(ump$layout, main = "UMAP plot, nbrs=4", xlab = "", ylab = "", col = gs, pch = 20, cex = 1.5)
# legend("topright",
#   inset = c(-0.15, 0), legend = levels(gs), pch = 20,
#   col = 1:nlevels(gs), title = "Group", pt.cex = 1.5
# )
# library("maptools") # point labels without overlaps
# pointLabel(ump$layout, labels = rownames(ump$layout), method = "SANN", cex = 0.6)
# # mean-variance trend, helps to see if precision weights are needed
plotSA(
  fit2,
  main = "Mean variance trend, Surget 2009"
)
```

```{r resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrusDefaultSettingsToIntegrationOriginalSpecie}
# Rename Limma columns to match the ones from DESeq2
setnames(
  x = tT2,
  old = c(
    "AveExpr",
    "logFC",
    "P.Value",
    "adj.P.Val"
  ),
  new = c(
    "baseMean",
    "log2FoldChange",
    "pvalue",
    "padj"
  )
)

project <- "Surget2009"
comparison <- "FluoxetineVsSalineForStressedInDentateGyrus"
tT2$project <- project
tT2$comparison <- comparison
dea_res[["microarray_v2"]][[project]][[comparison]] <- tT2[
  !(
    tT2$Gene.name %like% "," |
    is.na(tT2$Gene.name) |
    duplicated(tT2$Gene.name)
  ),
  cols_to_for_integration_original_specie
]
```

```{r resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrusDefaultSettingsToIntegration}
tT2 <- tT2[!is.na(tT2$mouse_symbol) %in% "", ]
# Probes are ordered by increasing p-value
# We keep only the one with the best pvalue for each gene
tT2 <- tT2[!duplicated(tT2$mouse_symbol), ]

best_orth <- gorth(
  tT2$mouse_symbol,
  source_organism = "mmusculus",
  target_organism = "hsapiens",
  mthreshold = 1,
  filter_na = FALSE
)

best_orth <- best_orth[
  ,
  !colnames(best_orth) %in% c(
    "input_number",
    "input_ensg",
    "ensg_number"
  )
]
setnames(
  best_orth,
  old = c(
    "ortholog_name",
    "ortholog_ensg"
  ),
  new = c(
    "hugo_symbol",
    "ensembl_gene_id"
  )
)

tT2 <- merge(
  tT2,
  best_orth,
  by.x = "mouse_symbol",
  by.y = "input",
  all.x = TRUE,
  sort = FALSE
)

tT2 <- tT2[!tT2$hugo_symbol %in% "N/A", ]
dea_res[["microarray"]][[project]][[comparison]] <- tT2
```


```{r resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrusDefaultSettingsHeatmapSettings}
fig_height <- 30
n_top_genes <- 200
gsubset <- gset[fData(gset)[["ID"]] %in% tT2[1:n_top_genes, ]$ID, ]
top_annotation_cols <- c(
  "stress",
  "treatment",
  "tissue"
)
```

```{r resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrusDefaultSettingsHeatmapRaw, dependson="resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrusDefaultSettingsHeatmapSettings", fig.height=fig_height}
set.seed(3)
Heatmap(
  exprs(gsubset),
  row_labels = fData(gsubset)$mouse_symbol,
  top_annotation = HeatmapAnnotation(
    df = pData(gsubset)[, top_annotation_cols]
  ),
  name = "RNA signal",
  column_title = "Samples",
  row_title = "Genes"
)
```

```{r resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrusDefaultSettingsHeatmapZscore, dependson="resultsMicroarraySurget2009FluoxetineVsSalineForStressedInDentateGyrusDefaultSettingsHeatmapSettings", fig.height=fig_height}
set.seed(3)
Heatmap(
  scaleRows(exprs(gsubset)),
  row_labels = fData(gsubset)$mouse_symbol,
  top_annotation = HeatmapAnnotation(
    df = pData(gsubset)[, top_annotation_cols]
  ),
  name = "RNA signal",
  column_title = "Samples",
  row_title = "Genes"
)
```

<!--This heatmap was used to check fold change sign is correct for this design:
We can see XXXXXX is overexpressed with Fluoxetine, which can be consistently seen on integration dotplots for the design `XXXX-FluoxetineVsControlForNoStressedInDentateGyrus`.
-->