#### Fluoxetine VS Untreated for stressed rats {#resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressed}

##### Default settings {#resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressedDefaultSettings}

Note that in the table below, "ENSEMBL" and "SYMBOL" correspond to newly annotated probesets, while "Gene.symbol" and "Gene.title" correspond to annotations updated by original authors in GEO.


```{r resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressedDefaultSettings}
gset <- getGEO(
  "GSE56028",
  destdir = geo_dir,
  GSEMatrix = TRUE,
  AnnotGPL = TRUE
)

if (length(gset) > 1) idx <- grep("GPL6247", attr(gset, "names")) else idx <- 1
gset <- gset[[idx]]

fData(gset) <- cbind(
  fData(gset),
  annotations$AFFY_RaGene_1_0_st_v1[
    match(
      fData(gset)$ID,
      annotations$AFFY_RaGene_1_0_st_v1$probe
    )
  ]
)

# Note the merge below is not equivalent to the cbind+match above as it is lead to mismatched probes and annotations for tT object
# fData(gset) <- merge(
#   fData(gset),
#   data.frame(annotations),
#   by.x = "ID",
#   by.y = "AFFY.RaGene.1.0.st.v1.probe",
#   sort = FALSE,
#   all.x = TRUE
# )

# # This dataset lack latest annotations for probes. Adding it here:
# annotations <- AnnotationDbi::select(
#   ragene10sttranscriptcluster.db,
#   #pd.ragene.1.0.st.v1,
#   keys = as.character(fData(gset)$ID),
#   keytype = "PROBEID",
#   columns = c(
#     'PROBEID',
#     'ENSEMBL',
#     'ENSEMBLTRANS',
#     'SYMBOL'
#   )
# )
# Ignoring duplicated annotations for now.
# Alternative strategy would be to "agregate()" duplicated rows
# to highlight probe associated to multiple genes.
# annotations <- annotations[!duplicated(annotations$PROBEID),]
# fData(gset) <- cbind(
#   fData(gset),
#   annotations
# )
# rm(annotations)

# make proper column names to match toptable
fvarLabels(gset) <- make.names(fvarLabels(gset))

gset <- gset[
  ,
  gset[["stress:ch1"]] %in% c(
    "unpredictable chronic mild stress"
  ) &
    gset[["agent:ch1"]] %in% c(
      "untreated",
      "fluoxetine"
    )
]

agent <- factor(make.names(gset[["agent:ch1"]]))

# log2 transformation
ex <- exprs(gset)
qx <- as.numeric(quantile(ex, c(0., 0.25, 0.5, 0.75, 0.99, 1.0), na.rm = T))
LogC <- (qx[5] > 100) ||
  (qx[6] - qx[1] > 50 && qx[2] > 0)
if (LogC) {
  ex[which(ex <= 0)] <- NaN
  exprs(gset) <- log2(ex)
}

design <- model.matrix(~ 0 + agent, gset)

fit <- lmFit(gset, design) # fit linear model

# set up contrasts of interest and recalculate model coefficients
contrast_matrix <- makeContrasts(
  contrasts = "agentfluoxetine-agentuntreated",
  levels = design
)

fit2 <- contrasts.fit(
  fit,
  contrast_matrix
)

# compute statistics and table of top significant genes
fit2 <- eBayes(
  fit2,
  0.01
)
tT <- topTable(
  fit2,
  adjust = "fdr",
  sort.by = "B",
  number = 250
)

tT <- subset(
  tT,
  select = c(
    "ID",
    "adj.P.Val",
    "P.Value",
    "t",
    "B",
    "logFC",
    #"ENSEMBL",
    #"SYMBOL",
    "Gene.symbol",
    "Gene.name",
    "Gene.title"
  )
)
produceDataTableWithButtons(tT)
# write.table(tT, file = stdout(), row.names = F, sep = "\t")

# Visualize and quality control test results.
# Build histogram of P-values for all genes. Normal test
# assumption is that most genes are not differentially expressed.
tT2 <- topTable(
  fit2,
  adjust = "fdr",
  sort.by = "B",
  number = Inf
)
hist(
  tT2$adj.P.Val,
  col = "grey",
  border = "white",
  xlab = "P-adj",
  ylab = "Number of genes",
  main = "P-adj value distribution"
)

# summarize test results as "up", "down" or "not expressed"
dT <- decideTests(
  fit2,
  adjust.method = "fdr",
  p.value = 0.05
)

# Venn diagram of results
vennDiagram(
  dT,
  circle.col = palette()
)

# create Q-Q plot for t-statistic
t.good <- which(!is.na(fit2$F)) # filter out bad probes
qqt(
  fit2$t[t.good],
  fit2$df.total[t.good],
  main = "Moderated t statistic"
)

# volcano plot (log P-value vs log fold change)
colnames(fit2) # list contrast names
ct <- 1 # choose contrast of interest
volcanoplot(
  fit2,
  coef = ct,
  main = colnames(fit2)[ct],
  pch = 20,
  highlight = length(which(dT[, ct] != 0)),
  names = rep("+", nrow(fit2))
)

# MD plot (log fold change vs mean log expression)
# highlight statistically significant (p-adj < 0.05) probes
plotMD(
  fit2,
  column = ct,
  status = dT[, ct],
  legend = FALSE,
  pch = 20,
  cex = 1
)
abline(h = 0)

################################################################
# General expression data analysis
ex <- exprs(gset)

# # box-and-whisker plot
# ord <- order(gs) # order samples by group
# palette(c(
#   "#1B9E77", "#7570B3", "#E7298A", "#E6AB02", "#D95F02",
#   "#66A61E", "#A6761D", "#B32424", "#B324B3", "#666666"
# ))
# par(mar = c(7, 4, 2, 1))
# title <- paste("GSE56028", "/", annotation(gset), sep = "")
# boxplot(ex[, ord], boxwex = 0.6, notch = T, main = title, outline = FALSE, las = 2, col = gs[ord])
# legend("topleft", groups, fill = palette(), bty = "n")

# # expression value distribution
# par(mar = c(4, 4, 2, 1))
# title <- paste("GSE56028", "/", annotation(gset), " value distribution", sep = "")
# plotDensities(ex, group = gs, main = title, legend = "topright")

# # UMAP plot (dimensionality reduction)
# ex <- na.omit(ex) # eliminate rows with NAs
# ex <- ex[!duplicated(ex), ] # remove duplicates
# ump <- umap(t(ex), n_neighbors = 9, random_state = 123)
# par(mar = c(3, 3, 2, 6), xpd = TRUE)
# plot(ump$layout, main = "UMAP plot, nbrs=9", xlab = "", ylab = "", col = gs, pch = 20, cex = 1.5)
# legend("topright",
#   inset = c(-0.15, 0), legend = levels(gs), pch = 20,
#   col = 1:nlevels(gs), title = "Group", pt.cex = 1.5
# )
# library("maptools") # point labels without overlaps
# pointLabel(ump$layout, labels = rownames(ump$layout), method = "SANN", cex = 0.6)

# mean-variance trend, helps to see if precision weights are needed
plotSA(
  fit2,
  main = "Mean variance trend, GSE56028"
)
```

```{r resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressedDefaultSettingsToIntegrationOriginalSpecie}
# Rename Limma columns to match the ones from DESeq2
setnames(
  x = tT2,
  old = c(
    "AveExpr",
    "logFC",
    "P.Value",
    "adj.P.Val"
  ),
  new = c(
    "baseMean",
    "log2FoldChange",
    "pvalue",
    "padj"
  )
)

project <- "GSE56028"
comparison <- "FluoxetineVsUntreatedForStressed"
tT2$project <- project
tT2$comparison <- comparison
dea_res[["microarray_v2"]][[project]][[comparison]] <- tT2[
  !(
    tT2$Gene.name %like% "," |
    is.na(tT2$Gene.name) |
    duplicated(tT2$Gene.name)
  ),
  cols_to_for_integration_original_specie
]
```

```{r resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressedDefaultSettingsToIntegration}
tT2 <- tT2[!tT2$Gene.symbol %in% "", ]
# Probes are ordered by increasing p-value
# We keep only the one with the best pvalue for each gene
tT2 <- tT2[!duplicated(tT2$Gene.symbol), ]

best_orth <- gorth(
  tT2$Gene.symbol,
  source_organism = "rnorvegicus",
  target_organism = "hsapiens",
  mthreshold = 1,
  filter_na = FALSE
)

best_orth <- best_orth[
  ,
  !colnames(best_orth) %in% c(
    "input_number",
    "input_ensg",
    "ensg_number"
  )
]
setnames(
  best_orth,
  old = c(
    "ortholog_name",
    "ortholog_ensg"
  ),
  new = c(
    "hugo_symbol",
    "ensembl_gene_id"
  )
)

tT2 <- merge(
  tT2,
  best_orth,
  by.x = "Gene.symbol",
  by.y = "input",
  all.x = TRUE,
  sort = FALSE
)

tT2 <- tT2[!tT2$hugo_symbol %in% "N/A", ]
dea_res[["microarray"]][[project]][[comparison]] <- tT2
```

```{r resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressedDefaultSettingsHeatmapSettings}
fig_height <- 30
n_top_genes <- 200
gsubset <- gset[fData(gset)[["ID"]] %in% tT2[1:n_top_genes, ]$ID, ]
```

```{r resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressedDefaultSettingsHeatmapRaw, dependson="resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressedDefaultSettingsHeatmapSettings", fig.height=fig_height}
set.seed(123)
Heatmap(
  exprs(gsubset),
  row_labels = fData(gsubset)$Gene.symbol,
  top_annotation = HeatmapAnnotation(
    df = pData(gsubset)[
      ,
      c(
        "stress:ch1",
        "tissue:ch1",
        "agent:ch1"
      )
    ]
  ),
  name = "RNA signal",
  column_title = "Samples",
  row_title = "Genes"
)
```

```{r resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressedDefaultSettingsHeatmapZscore, dependson="resultsMicroarrayGSE56028FluoxetineVsUntreatedForStressedDefaultSettingsHeatmapSettings", fig.height=fig_height}
set.seed(123)
Heatmap(
  scaleRows(exprs(gsubset)),
  row_labels = fData(gsubset)$Gene.symbol,
  #row_labels = tT2[1:n_top_genes, ]$hugo_symbol,
  top_annotation = HeatmapAnnotation(
    df = pData(gsubset)[
      ,
      c(
        "stress:ch1",
        "tissue:ch1",
        "agent:ch1"
      )
    ]
  ),
  name = "RNA signal",
  column_title = "Samples",
  row_title = "Genes"
)
```

<!--This heatmap was used to check fold change sign is correct for this design:
We can see XXXXXX is overexpressed with Fluoxetine, which can be consistently seen on integration dotplots for the design `XXXX-FluoxetineVsControlForNoStressedInDentateGyrus`.
-->