### SRP131063 {#resultsRnaseqSRP131063}

```{r resultsRnaseqSRP131063}
smi <- "out/r/tidy_featureCounts/subread/featureCounts_-O_-t_exon_-g_merge_gene_id_name_gtf-Rnor6-merge-attr-retrieve-ensembl_bam-Rnor6-SRP131063_raw.tsv"
linkSmiToRmdAndBookDir(smi)
SRP131063_raw <- fread(smi)
produceDataTableWithButtons(head(SRP131063_raw))

smi <- "out/py/retrieve_sra_metadata/SRP131063.xlsx"
linkSmiToRmdAndBookDir(smi)
SRP131063_metadata <- read_excel(smi)
produceDataTableWithButtons(SRP131063_metadata)

SRP131063_genes <- SRP131063_raw[, 1]
SRP131063_genes[
  ,
  c(
    "rat_ensembl_gene_id",
    #"rat_symbol"
    "Gene.name"
  ) := tstrsplit(
    Geneid,
    "|",
    fixed = TRUE
  )
]
# Some ENSEMBLE gene have "." as gene.name
SRP131063_genes[
  Gene.name %like% "^\\.$",
  Gene.name := rat_ensembl_gene_id
]
SRP131063_genes[, Geneid := NULL]

if (exists("rat_to_human_best_orth")) {
  message("The mapping from Rat to Human gene has already been queried to g:Orth in a previous section. The same table is reused.")
} else {
  # rat_to_human_orths <- gorth(
  #   SRP131063_genes$rat_ensembl_gene_id,
  #   source_organism = "rnorvegicus",
  #   target_organism = "hsapiens",
  #   filter_na = FALSE
  # )
  rat_to_human_best_orth <- gorth(
    SRP131063_genes$rat_ensembl_gene_id,
    source_organism = "rnorvegicus",
    target_organism = "hsapiens",
    mthreshold = 1,
    filter_na = FALSE
  )

  # There is only a limited number of multiple orthologues:
  # > dim(rat_to_human_orths)
  # [1] 36521     7
  # > dim(rat_to_human_best_orth)
  # [1] 32620     7

  rat_to_human_best_orth <- rat_to_human_best_orth[
    ,
    !colnames(rat_to_human_best_orth) %in% c(
      "input_number",
      "input_ensg",
      "ensg_number"
    )
  ]
  setnames(
    rat_to_human_best_orth,
    old = c(
      "ortholog_name",
      "ortholog_ensg"
    ),
    new = c(
      "hugo_symbol",
      "ensembl_gene_id"
    )
  )
}

SRP131063_genes <- merge(
  SRP131063_genes,
  rat_to_human_best_orth,
  by.x = "rat_ensembl_gene_id",
  by.y = "input",
  all.x = TRUE,
  sort = FALSE
)

row.names(SRP131063_metadata) <- SRP131063_metadata$accession

SRP131063_metadata$condition <- ifelse(
  SRP131063_metadata$source_name %like% "Rats exposed to chronic stress",
  "stress",
  "control"
)
SRP131063_metadata$treatment <- ifelse(
  SRP131063_metadata$source_name %like% "saline",
  "saline",
  ifelse(
    SRP131063_metadata$source_name %like% "fluoxetine",
    "fluoxetine",
    ifelse(
      SRP131063_metadata$source_name %like% "albiflorin",
      "albiflorin",
      "THERE IS AN ERROR IN THE CODE IF THIS CONDITION IS SATISFIED"
    )
  )
)
SRP131063_metadata$group <- paste(
  SRP131063_metadata$condition,
  SRP131063_metadata$treatment,
  sep = "."
)
SRP131063_metadata$group <- factor(
  SRP131063_metadata$group,
  levels = c(
    "stress.saline",
    "stress.fluoxetine",
    "stress.albiflorin",
    "control.saline"
  )
)
```
